Check alternate chains prior to default chain when providing force_chain

oskgu360 opened the pull request:
This allows for retrieving the alternate chain when both default and alternate contain the provided issuer

### WHY are these changes introduced?

To get rid of misleading upstream in repo https://github.com/84codes/acme-client

### WHAT is this pull request doing?

Applies same patch as used in https://github.com/84codes/acme-client

dentarg commented:
This is great @jage but I want this fork to be a fork of https://github.com/unixcharles/acme-client and not https://github.com/oskgu360/acme-client (if we still need to use a fork)

dentarg commented:
That has been on my private to do list for quite some time... :)

oskgu360 commented:
Ah, missed this. I'll try to see if its still needed asap, just gotta read up on the details again hehe

oskgu360 reviewed (approved):
Tried generating a cert without our custom changes, and we do then get the LE default chain which we do not want. So to keep compatibility with older openssl clients (Which I assume we have a few of), we need to keep this custom changes in the fork.

For context: 
https://community.letsencrypt.org/t/long-default-and-short-alternate-certificate-chains-explained/162526

Don't see why this PR would change our behaviour though, so LGTM :)

oskgu360 commented:
The cross-signing of `ISRG Root X1` by `DST Root CA X3` expires on
`Not After : Sep 30 18:14:03 2024 GMT`, my guess is that after this, LE will make the chain we seek default 🤞

dentarg commented:
Let's work in the direct fork? https://github.com/84codes/acme-client-2

oskgu360 committed:
Check alternate chains prior to default chain when providing force_chain

This allows for retrieving the alternate chain when both default and alternate contain the provided issuer

dentarg commented:
I think we should keep our change in our branch and rebase the branch with master and keep master same as upstream master.

oskgu360 commented:
okok!
